elastic/synthetics-recorder
# chore: update dependency @elastic/synthetics and Playwright

closes #262 and #263
relates to https://github.com/elastic/playwright/pull/1

## Summary
This PR updates a couple of dependencies, the latest [@elastic/synthetics](https://github.com/elastic/synthetics) agent (0.0.1-beta.31) and playwright version that the agent is using.
This update is especially exciting as Synthetics Recorder now generates scripts with [Locator API](https://playwright.dev/docs/api/class-locator) 🎉 

Some notable changes:
- Uses Node version to 16.15.0, npm >= 7 (package-lock.json is now version 2) 
- Removes `npm-force-resolutions` preinstall hook as it pollutes the npm install behaviour(since npm@7)
- Adds temporary type definition around `ActionInContext` and `Step` to make it compatible with the newer Playwright. These type definitions will be deleted once synthetics agent exports the compatible type definition again
- After updating Node version, we needed to build a new Docker image. With the new image and the new node, CI was having some issues with the permission when executing npm scripts. It seems like the docker tries to run the node executable installed by the host workspace that is mounted as volume. As a workaround, we install node dependencies rather than mounting the workspace volume when building the image.


## How to validate this change
Pull this branch and run `npm ci` and run `npm run dev`. Confirm that everything works as before, also note that the generated code now contains `page.locator(...)`.

## TODO
-  ~Find out what has changed to the signal url when recording the script. It's making the test fail as it waits for the navigation to the URL with the specific querystring which usually generated randomly on each navigation.~
**edit**: _The same behaviour is observed in the main branch as well so thus it is not related to the playwright version update. will address it separately

## Next Steps
- Test releases
- Delegate the code generation to Synthetics agent
